### DISCUSSION TITLE
How can I start qutebrowser with --temp-basedir but inject my actual history, bookmarks and quickmarks?

### DISCUSSION THREAD

**Discussion by Aster89** [qutebrowser/qutebrowser#7850] [Q&A] [answered]
The reason being that after the last update (to Qt6) I find that when I type keys after `:open [-t] `, the reaction of the filtering is extremely slow and it eats so much RAM that the vent starts spinning more loudly.

In order to cut my settings out of the equation I want to try reproducing the issue with `--temp-basedir`, but without a long history (+ book/quickmarks) to filter through, it's obvious that I don't see any slowdown.

**Comment by toofar** [comment]
It would probably be easier to make an actual basedir (that you can delete later!) to test. That way you can copy things across as needed.

**Reply by Aster89** [reply]
Oh, so I guess I just need to use the `--basedir BASEDIR` option, instead of `--temp-basedir`. But which directory is `BASEDIR` supposed to play the role of? I've tried making a copy of `$HOME/.config/qutebrowser` and then starting `qutebrowser --basedir aCopyOfThatDir`, but it doesn't seem like it made use of it.

**Reply by emanuele6** [reply, marked as answer]
@Aster89 

If you use `/tmp/foo` as basedir, then the config path (equivalent of `~/.config/qutebrowser`) will be `/tmp/foo/config`.
So, if you want to use the same configuration files, you would copy the files in `~/.config/qutebrowser` into `/tmp/foo/config`, and then run qutebrowser with `-B /tmp/foo` (or `--basedir`, and or using some other directory instead of `/tmp/foo`).

Also note that you can always check what directories qutebrowser uses in the `:version`/`qute://version` page:
![file](https://github.com/qutebrowser/qutebrowser/assets/20175435/9f8776b9-2ed6-4813-9db3-b4d3c10dda88)

(left: my standard qutebrowser; right: `-B /tmp/foo`)

**Reply by emanuele6** [reply]
Your history is stored in the "data" directory.
IIRC copying the "data" directory while qutebrowser is running can cause problems because some of the files are sqlite3 databases, and they will be locked by `qutebrowser` when it is running; so you could copy a locked databases that the new qutebrowser cannot use.
I suggest closing qutebrowser, and making sure it is not running, before copying files in the data directory to avoid problems.
